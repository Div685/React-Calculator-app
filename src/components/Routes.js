import React from 'react';
import { BrowserRouter, Switch, Route } from 'react-router-dom';
import Home from './Home';
import NavBar from './NavBar';
import Quote from './Quote';
import App from './App';

function Routes() {
  return (
    <BrowserRouter>
      <NavBar />
      <Switch>
        <Route exact path="/" component={Home} />
        <Route exact path="/calculator" component={App} />
        <Route exact path="/quote" component={Quote} />
      </Switch>
    </BrowserRouter>
  );
}

export default Routes;
