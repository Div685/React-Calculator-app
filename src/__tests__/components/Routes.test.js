import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import Routes from '../../components/Routes';

describe('Routes Component', () => {
  it('renders Routes page correctly', () => {
    const tree = renderer.create(<Routes />);
    expect(tree).toMatchSnapshot();
  });
});

describe('Test Routes Component for NavBar Links', () => {
  it('it test the home page', () => {
    render(<Routes />);
    expect(screen.getByTestId('page-wrapper')).toHaveTextContent('Welcome to our Page!');
  });

  it('Test the Calculator Page if the Calculator Link in the NavBar is clicked', () => {
    render(<Routes />);
    fireEvent.click(screen.getByText('Calculator'));
    expect(screen.getByTestId('page-wrapper')).toHaveTextContent('Let\'s do some math!');
  });
});
