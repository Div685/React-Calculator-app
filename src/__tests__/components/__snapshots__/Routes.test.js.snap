// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes Component renders Routes page correctly 1`] = `
Array [
  <div
    className="nav"
  >
    <nav>
      <div
        className="title"
      >
        <h1>
          <a
            href="/"
            onClick={[Function]}
          >
            Math Magicians
          </a>
        </h1>
      </div>
      <div
        className="navLink"
      >
        <div>
          <a
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </div>
        <div>
          <a
            href="/calculator"
            onClick={[Function]}
          >
            Calculator
          </a>
        </div>
        <div>
          <a
            href="/quote"
            onClick={[Function]}
          >
            Quote
          </a>
        </div>
      </div>
    </nav>
  </div>,
  <div
    data-testid="page-wrapper"
  >
    <div
      className="home"
    >
      <h1>
        Welcome to our Page!
      </h1>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui o!cia deserunt mollit anim id est laborum. Ipsu
      </p>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui o!cia deserunt mollit anim id est laborum. Ipsu
      </p>
    </div>
  </div>,
]
`;
